### PR TITLE
Can use Ctrl-X to open code using $EDITOR

### PR DESCRIPTION
**Pull request by Tangdongle**
Added support for editing code using your environment's $EDITOR

Works, but I had to extend nim-noise to add an extra meta character (https://github.com/jangko/nim-noise/pull/9). Can wait for that to be merged and I'll revert the nimble require here: https://github.com/inim-repl/INim/compare/master...TangersTongle:editor_support?expand=1#diff-0fdaf9697c4d59b94f10d5a02de044c3R16

### CODE REVIEW

**Comment by 0atman** [comment]
Wow! That's a great feature!

**Comment by 0atman** [comment]
OK great, I'm really pleased to see your changes merged into Noise.

We should test this thoroughly on different systems, as this interacts in a difficult to predict way with the OS.

Separately to this bahaviour issue, I'm getting a syntax error:
`/home/oatman/projects/INim/src/inim.nim(313, 8) Error: undeclared identifier: 'ktCtrlX'`

Which I'm suspicious of because the CI tests didn't catch it... Is it just me? Does it happen for you when you merge the latest master into this PR?

**Comment by Tangdongle** [comment]
Hmmm, @0atman does it still fail if you run `nimble uninstall noise` and then `nimble install -d` in the INim directory? It sounds like your copy of noise doesn't have the commit I made merged in. Maybe it's something we need to flag in the nimble file?

**Comment by Tangdongle** [comment]
I agree with the testing, I have a windows VM to test in and I know a guy who has OSX, they might be able to give it a few run-throughs

**Comment by 0atman** [comment]
Ah, that's it. Old version of noise 👌

**Comment by Tangdongle** [comment]
Maybe we can pin the dep against the current version so it gets pulled in when people run `nimble update inim`

**Comment by 0atman** [comment]
```
λ  INim git:(TangersTongle-editor_support) ✗ nimble uninstall inim
    Looking for inim (any version)
   Checking reverse dependencies
    Prompt: The following packages will be removed:
  inim (0.4.7), inim (0.5.0)
Do you wish to continue? [y/N]
    Answer: y     
    Removed inim (0.4.7)
    Removed inim (0.5.0)
 λ  INim git:(TangersTongle-editor_support) ✗ nimble uninstall noise
    Looking for noise (any version)
   Checking reverse dependencies
    Prompt: The following packages will be removed:
  noise (0.1.3)
Do you wish to continue? [y/N]
    Answer: y
    Removed noise (0.1.3)
 λ  INim git:(TangersTongle-editor_support) ✗ nimble build
  Verifying dependencies for inim@0.5.0
      Info: Dependency on cligen@>= 0.9.15 already satisfied
  Verifying dependencies for cligen@0.9.46
 Installing noise@any version
Downloading https://github.com/jangko/nim-noise using git
  Verifying dependencies for noise@0.1.3
 Installing noise@0.1.3
   Success: noise installed successfully.
   Building inim/inim using c backend
```

That fix indeed worked. But this represents an upgrade path problem, right? Would pinning the version as you say fix that for upgraders?

File: src/inim.nim [hunk 1]
```diff
@@ -474,6 +492,8 @@ proc main(nim = "nim", srcFile = "", showHeader = true,
     indentSpaces = ""
     sessionNoAutoIndent = noAutoIndent
 
+  app.editor = getEnv("EDITOR")
```
Comment (0atman) [inline src/inim.nim:495]: This doesn't work for me:

```
nim> getEnv "EDITOR"
 == type TaintedString
nim> 

Quitting INim: Goodbye!
 λ echo $EDITOR
nvim
```

Why is nim not picking up `$EDITOR` on my machine I wonder...

Comment (0atman) [reply src/inim.nim:495]:
```
./inim
👑 INim 0.5.0
Nim Compiler Version 1.2.0 [Linux: amd64] at /home/oatman/.nimble/bin/nim
nim> 
No $EDITOR set in ENV
```

Comment (Tangdongle) [reply src/inim.nim:495]: Hmmm, what do you get from `echo $EDITOR; ./inim`? And does `EDITOR=vim ./inim` correctly set EDITOR to `vim`?

Comment (0atman) [reply src/inim.nim:495]: Oh wow, my fish shell is broken sorry! This works great in zsh :laughing:

**Comment by 0atman** [comment]
OK, works great on Ubuntu 20.04 - how is it on Windows?

**Comment by Tangdongle** [comment]
Yeah, I believe we can pin by commit hash. Not sure if we can do >= than a particular hash.

Will test on windows tomorrow when I have my vbox available

**Comment by Tangdongle** [comment]
@0atman I checked this on Windows. I tested with Notepad.exe as my editor and set EDITOR on environment variables (set to %windir\system32\notepad.exe"). Also tested with a downloaded editor (vim) and that also worked for me.

**Pull request merged**